build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/JdkInfo.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xamarin.Android.Build.Utilities;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public class JdkInfo : Task
+	{
+		[Required]
+		public ITaskItem Output { get; set; }
+
+		public string AndroidNdkPath { get; set; }
+
+		public string AndroidSdkPath { get; set; }
+
+		public string JavaSdkPath { get; set; }
+
+		public override bool Execute ()
+		{
+			Log.LogMessage (MessageImportance.Low, $"Task {nameof (JdkInfo)}");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Output)}: {Output}");
+
+			AndroidLogger.Error += ErrorHandler;
+			AndroidLogger.Warning += WarningHandler;
+			AndroidLogger.Info += InfoHandler;
+			try {
+				AndroidSdk.Refresh (AndroidSdkPath, AndroidNdkPath, JavaSdkPath);
+
+				var javaSdkPath = AndroidSdk.JavaSdkPath;
+				Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidSdk.JavaSdkPath)}: {javaSdkPath}");
+
+				var jvmPath = Path.Combine (javaSdkPath, "jre", "bin", "server", "jvm.dll");
+				if (!File.Exists (jvmPath)) {
+					Log.LogError ($"JdkJvmPath not found at {jvmPath}");
```

Comment:
...alternatively, if we *require* that `jvmPath` exist, then it follows that we should add `[Required]` to `JavaSdkPath`, and let MSBuild do the null/empty check for us.